extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheMethodValidator.java

```diff
@@ -0,0 +1,115 @@
+            List<DeploymentException> exceptions) {
+        boolean found = false;
+        AnnotationInstance annotation = annotationStore.getAnnotation(method, CACHE_LOAD);
+        if (annotation != null) {
+            if (method.returnType().kind() == Type.Kind.VOID) {
+                exceptions.add(new DeploymentException(
+                        CACHE_LOAD + " annotation is not allowed on a method returning void: [class: "
+                                + bean.getBeanClass() + ", method: " + method + "]"));
+            } else {
+                found = true;
+                validateCacheName(bean, annotation, exceptions);
+            }
+        }
+        return found;
+    }
+
+    private static boolean validateCacheStoreAnnotation(AnnotationStore annotationStore, BeanInfo bean, MethodInfo method,
+            List<DeploymentException> exceptions) {
+        boolean found = false;
+        AnnotationInstance cacheStoreAnnotation = annotationStore.getAnnotation(method, CACHE_STORE);
+        if (cacheStoreAnnotation != null) {
+            if (method.returnType().kind() == Type.Kind.VOID) {
+                exceptions.add(new DeploymentException(
+                        CACHE_STORE + " annotation is not allowed on a method returning void: [class: "
+                                + bean.getBeanClass() + ", method: " + method + "]"));
+            } else {
+                found = true;
+                validateCacheName(bean, cacheStoreAnnotation, exceptions);
+            }
+        }
+        return found;
+    }
+
+    private static boolean validateCacheInvalidateAnnotation(AnnotationStore annotationStore, BeanInfo bean, MethodInfo method,
+            List<DeploymentException> exceptions) {
+        boolean found = false;
+        AnnotationInstance cacheInvalidateAnnotation = annotationStore.getAnnotation(method, CACHE_INVALIDATE);
+        if (cacheInvalidateAnnotation != null) {
+            found = true;
+            validateCacheName(bean, cacheInvalidateAnnotation, exceptions);
+        }
+        return found;
+    }
+
+    private static boolean validateCacheInvalidateAllAnnotation(AnnotationStore annotationStore, BeanInfo bean,
+            MethodInfo method, List<DeploymentException> exceptions) {
+        boolean found = false;
+        AnnotationInstance cacheInvalidateAnnotation = annotationStore.getAnnotation(method, CACHE_INVALIDATE_ALL);
+        if (cacheInvalidateAnnotation != null) {
+            found = true;
+            validateCacheName(bean, cacheInvalidateAnnotation, exceptions);
+        }
+        return found;
+    }
+
+    private static void validateCacheName(BeanInfo bean, AnnotationInstance annotation, List<DeploymentException> exceptions) {
+        AnnotationValue annotationValue = annotation.value(CACHE_NAME_PARAMETER_NAME);
+        if (annotationValue != null && annotationValue.asString().startsWith(GENERATED_CACHE_NAME_PREFIX)) {
```

Comment:
For now, generated cache names are prefixed with `generated_` and it is forbidden at build time to use the same prefix for a user-chosen cache name.